### DISCUSSION TITLE
API is invalid! What am I doing wrong here?

### DISCUSSION THREAD

**Discussion by STROUDICOUS** [brianpetro/obsidian-smart-connections#478] [General]
I've definitely got the key from openAI. sk-FjHtUXXXXXXXXXXXXXXXXXXXXXXXXXXXXX1gXNX8NNxxXXXXX is the format.

I'm sure I've done this correctly. What else might be causing the error?
When I press the "SAVE" button on the OpenAI API Key section of the plug options pages I get...

[smart connections] Error: API is invalid!

**Comment by brianpetro** [comment]
@STROUDICOUS it most likely has to do with your OpenAI account. Check out this thread for suggestions https://github.com/brianpetro/obsidian-smart-connections/discussions/155#discussioncomment-7632050

**Reply by STROUDICOUS** [reply]
Thanks you. Gotchya. API usage is billed separately. If I become a smart connections supporter (I'm also a Plus subscriber),  do I still need to top up the API pay as you go? Or perhaps there's different functionality depending on my settings and subscriptions?

**Reply by brianpetro** [reply]
Being a Smart Connections supporter will give you access to the ChatGPT integrations that also require a ChatGPT Plus subscription. 

You could use that ChatGPT Integration instead of the Smart Chat, which currently relies on the OpenAI API, within Obsidian.

Besides that, you could use local embedding models instead of OpenAI embeddings to reduce API usage, but this will likely be a negligible impact compared to the Smart Chat API usage.

🌴
